integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/NextMoveServiceBus.java

```diff
@@ -1,87 +1,120 @@
+import org.springframework.util.StringUtils;
 
 import javax.annotation.PostConstruct;
-import javax.xml.bind.*;
-import javax.xml.transform.stream.StreamSource;
-import java.io.ByteArrayInputStream;
+import javax.xml.bind.JAXBException;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.time.Duration;
 import java.util.ArrayList;
+import java.util.Base64;
 import java.util.Collection;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 
-import static com.google.common.base.Strings.isNullOrEmpty;
 import static java.lang.String.format;
-import static java.util.Arrays.asList;
-import static no.difi.meldingsutveksling.ServiceIdentifier.DPE_DATA;
-import static no.difi.meldingsutveksling.ServiceIdentifier.DPE_INNSYN;
-import static no.difi.meldingsutveksling.nextmove.ServiceBusQueueMode.DATA;
-import static no.difi.meldingsutveksling.nextmove.ServiceBusQueueMode.INNSYN;
-import static no.difi.meldingsutveksling.nextmove.ServiceBusQueueMode.MEETING;
+import static no.difi.meldingsutveksling.NextMoveConsts.ASIC_FILE;
+import static no.difi.meldingsutveksling.ServiceIdentifier.DPE;
+import static no.difi.meldingsutveksling.nextmove.ConversationDirection.INCOMING;
+import static no.difi.meldingsutveksling.nextmove.ServiceBusQueueMode.*;
 
 @Component
+@ConditionalOnProperty(name = "difi.move.feature.enableDPE", havingValue = "true")
+@Slf4j
 public class NextMoveServiceBus {
 
-    private static final Logger log = LoggerFactory.getLogger(NextMoveServiceBus.class);
+    private final IntegrasjonspunktProperties props;
+    private final NextMoveQueue nextMoveQueue;
+    private final ServiceBusRestClient serviceBusClient;
+    private final ObjectMapper om;
+    private final MessageContextFactory messageContextFactory;
+    private final AsicHandler asicHandler;
+    private final InternalQueue internalQueue;
+    private final CryptoMessagePersister cryptoMessagePersister;
+    private final ServiceBusPayloadConverter payloadConverter;
+    private final SBDReceiptFactory sbdReceiptFactory;
+    private final ServiceRegistryLookup serviceRegistryLookup;
+    private final ConversationService conversationService;
+    private final MessageStatusFactory messageStatusFactory;
+    private final TimeToLiveHelper timeToLiveHelper;
+    private final SBDUtil sbdUtil;
+    private final TaskExecutor taskExecutor;
 
-    private static final String NEXTMOVE_QUEUE_PREFIX = "nextbestqueue";
-
-    private IntegrasjonspunktProperties props;
-    private StandardBusinessDocumentFactory sbdf;
-    private MessageSender messageSender;
-    private NextMoveQueue nextMoveQueue;
-    private JAXBContext jaxbContext;
-    private ServiceBusRestClient serviceBusClient;
     private IMessageReceiver messageReceiver;
-    private InternalQueue internalQueue;
 
-    @Autowired
     public NextMoveServiceBus(IntegrasjonspunktProperties props,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 16 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00107)